src/lightning/pytorch/core/saving.py

```diff
@@ -59,6 +60,8 @@ def _load_from_checkpoint(
     **kwargs: Any,
 ) -> Union["pl.LightningModule", "pl.LightningDataModule"]:
     with pl_legacy_patch():
+        if map_location is None:
+            map_location = "cpu" if not CUDAAccelerator.is_available() else None
```

Comment:
as a oneliner we could also write 

```suggestion
            map_location = "cpu" if not CUDAAccelerator.is_available() else None if map_location is None else map_location
```